transport-native-epoll/src/main/java/io/netty/channel/epoll/LinuxSocket.java

```diff
@@ -0,0 +1,162 @@
+    }
+
+    void setTcpKeepIntvl(int seconds) throws IOException {
+        setTcpKeepIntvl(intValue(), seconds);
+    }
+
+    void setTcpKeepCnt(int probes) throws IOException {
+        setTcpKeepCnt(intValue(), probes);
+    }
+
+    void setTcpUserTimeout(int milliseconds) throws IOException {
+        setTcpUserTimeout(intValue(), milliseconds);
+    }
+
+    void setIpFreeBind(boolean enabled) throws IOException {
+        setIpFreeBind(intValue(), enabled ? 1 : 0);
+    }
+
+    void getTcpInfo(EpollTcpInfo info) throws IOException {
+        getTcpInfo(intValue(), info.info);
+    }
+
+    void setTcpMd5Sig(InetAddress address, byte[] key) throws IOException {
+        final NativeInetAddress a = NativeInetAddress.newInstance(address);
+        setTcpMd5Sig(intValue(), a.address(), a.scopeId(), key);
+    }
+
+    boolean isTcpCork() throws IOException  {
+        return isTcpCork(intValue()) != 0;
+    }
+
+    int getTcpDeferAccept() throws IOException {
+        return getTcpDeferAccept(intValue());
+    }
+
+    boolean isTcpQuickAck() throws IOException {
+        return isTcpQuickAck(intValue()) != 0;
+    }
+
+    long getTcpNotSentLowAt() throws IOException {
+        return getTcpNotSentLowAt(intValue()) & MAX_UINT32_T;
+    }
+
+    int getTcpKeepIdle() throws IOException {
+        return getTcpKeepIdle(intValue());
+    }
+
+    int getTcpKeepIntvl() throws IOException {
+        return getTcpKeepIntvl(intValue());
+    }
+
+    int getTcpKeepCnt() throws IOException {
+        return getTcpKeepCnt(intValue());
+    }
+
+    int getTcpUserTimeout() throws IOException {
+        return getTcpUserTimeout(intValue());
+    }
+
+    boolean isIpFreeBind() throws IOException {
+        return isIpFreeBind(intValue()) != 0;
+    }
+
+    PeerCredentials getPeerCredentials() throws IOException {
+        return getPeerCredentials(intValue());
+    }
+
+    public static LinuxSocket newSocketStream() {
+        return new LinuxSocket(newSocketStream0());
+    }
+
+    public static LinuxSocket newSocketDgram() {
+        return new LinuxSocket(newSocketDgram0());
+    }
+
+    public static LinuxSocket newSocketDomain() {
+        return new LinuxSocket(newSocketDomain0());
+    }
+
+    private static native int getTcpDeferAccept(int fd) throws IOException;
+    private static native int isTcpQuickAck(int fd) throws IOException;
+    private static native int isTcpCork(int fd) throws IOException;
+    private static native int getTcpNotSentLowAt(int fd) throws IOException;
+    private static native int getTcpKeepIdle(int fd) throws IOException;
+    private static native int getTcpKeepIntvl(int fd) throws IOException;
+    private static native int getTcpKeepCnt(int fd) throws IOException;
+    private static native int getTcpUserTimeout(int fd) throws IOException;
+    private static native int isIpFreeBind(int fd) throws IOException;
+    private static native void getTcpInfo(int fd, int[] array) throws IOException;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Consider using varargs for methods or constructors which take an array the last parameter. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3AUseVarargs)